### DISCUSSION TITLE
一つのキーを押すと3字くらい同時に反応する

### DISCUSSION THREAD

**Discussion by aranea0902** [Yowkees/keyball#292] [Q&A] [answered]
keyball61でprintscreenのキーを押すとハイフン、バックスラッシュ、アポストロフィーが入力されます
トラックボールをつけられる基盤の一番右下のキーです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

１番右下のキーというのはどのキーになりますでしょうか？トラックボールに近いキーですか？

できればRemapのKey Matrix testにて、同時入力されてしまうキーをご確認いただき、スクリーンショットをお送りいただけますと幸いです。

お手数ですがご確認お願いいたします。

**Comment by aranea0902** [comment]
![スクリーンショット 2023-07-31 113117](https://github.com/Yowkees/keyball/assets/140910521/7ac751dd-a80b-4902-8650-ff0ce7d36fda)
赤で丸したところを押すと同じ列が動作しているようです。

**Comment by Yowkees** [comment]
症状としては、ダイオードの向きが間違っているものがあるように思います。
特にLED22の近くのダイオードとLED12の近くのダイオードが関係しそうですが、確かではないので、これらを中心に全体的に方向を確認いただければ幸いです。
お手数ですがご確認お願いいたします。

**Comment by aranea0902** [comment]
led22,12周辺のダイオードの向きだけでなくすべてのダイオードを確認しましたが特に間違っているものはなさそうでした
一応12，22近くのダイオードを付け替えましたが症状は変わりません

**Comment by Yowkees** [comment]
ご対応ありがとうございました。
あとはProMicro裏の半田付けで隣とショートしているものがないかご確認ください。ここはスルーホールがジグザグに配置されていて隣と近くショートしやすいです。
もし改善されなければ問題の基板の半田付け面のお写真を見せていただければ幸いです。
以上、ご検討よろしくお願いいたします。

**Comment by aranea0902** [comment]
一応確認したのですが自分にはわかりませんでした。はんだ付け初心者なのでかなり下手ですが写真を挙げておきます
![IMG_0224](https://github.com/Yowkees/keyball/assets/140910521/741420d0-7c7b-4958-8c4b-b6118020529d)

**Comment by Yowkees** [comment]
お写真ありがとうございます。それほど問題となりそうな個所は写真からは見当たりませんね。
左右のProMicroを入れ替えるとどうなるでしょうか？

**Comment by aranea0902** [comment]
左右のpromicroを入れ替えても変わりません

**Comment by Yowkees** [comment]
了解しました。
一番右下のキーを押すと、縦一列すべて入力されますが、その他の右端列のキーが押された場合は、どのような挙動になりますか？

**Reply by aranea0902** [reply]
一番右下のキー以外はそれぞれ問題なく入力されます

**Comment by Yowkees** [comment, marked as answer]
なるほど。そうなりますと、やはり右下キー周辺に問題がありそうですね。

ひとつ確認いただきたいのが、右下キーのキーソケットの端子が他の線に触れていないかです。

下記画像の赤丸部分、はんだの量が多いので半田吸い取り線等を使用して、端子外への膨らみを除去してみていただきたいです。
もしかするとその下のＧＮＤ等に接触しているかも知れません。
通常、周辺の導線にはレジストがかかっているので半田はつきませんが、キズがついていたりするとショートする場合があります。

以上、お手数ですがご対応お願いいたします。

![KB61_keyfix2](https://github.com/Yowkees/keyball/assets/84209772/ff97bc3c-0de3-470e-ac01-a308d64ceaeb)

**Comment by aranea0902** [comment]
左のほうのはんだがきずついてむき出しになった導線についてショートしてましたはんだを吸って付け直したら治りました
ありがとうございました。

**Comment by Yowkees** [comment]
無事に直ったようでよかったです！ご対応ありがとうございました。
ぜひKeyball61を末永くご愛用いただければ幸いです。
